spec/apollo-federation/service_field_v2_spec.rb

```diff
@@ -128,6 +128,98 @@ def execute_sdl(schema)
       )
     end
 
+    it 'returns the federation SDL with multiple links for the schema' do
+      product = Class.new(base_object) do
+        graphql_name 'Product'
+
+        field :upc, String, null: false
+      end
+
+      query_obj = Class.new(base_object) do
+        graphql_name 'Query'
+
+        field :product, product, null: true
+      end
+
+      schema = Class.new(base_schema) do
+        query query_obj
+        federation version: '2.3',
+                   links: [{
+                     url: 'https://specs.example.com/federation/v2.3',
+                     import: ['test'],
+                   }]
+      end
+
+      expect(execute_sdl(schema)).to match_sdl(
+        <<~GRAPHQL,
+          extend schema
+            @link(url: "https://specs.apollo.dev/federation/v2.3", import: ["@inaccessible", "@tag", "@composeDirective"])
+            @link(url: "https://specs.example.com/federation/v2.3", import: ["@test"])
+
+          type Product {
+            upc: String!
+          }
+
+          type Query {
+            product: Product
+          }
+        GRAPHQL
+      )
+    end
+
+    it 'returns the federation SDL with compose directives for the schema' do
+      complexity_directive = Class.new(GraphQL::Schema::Directive) do
+        graphql_name 'complexity'
+        argument :fixed, Integer, required: true
+        description 'complexity of the field'
+        locations GraphQL::Schema::Directive::FIELD_DEFINITION
+      end
+
+      product = Class.new(base_object) do
+        graphql_name 'Product'
+
+        field :upc, String, null: false
+      end
+
+      query_obj = Class.new(base_object) do
+        graphql_name 'Query'
+
+        field :product, product, null: true, directives: { complexity_directive => { fixed: 1 } }
+      end
+
+      schema = Class.new(base_schema) do
+        query query_obj
+        federation version: '2.3',
+                   links: [{
+                     url: 'https://specs.example.com/federation/v2.3',
+                     import: [complexity_directive.graphql_name],
```

Comment:
This one probably doesn't need importing with an @link since it is being defined inline.

Reply:
It looks like even if it's being defined inline it needs to be referenced via a custom `link`. For example, if I try to compose a supergraph with an inline directive that isn't imported with `link` I get:
>    UNKNOWN: Directive "@..." in subgraph "..." cannot be composed because it is not a member of a core feature 

In the [docs](https://www.apollographql.com/docs/federation/federated-types/federated-directives/#composedirective) they're doing it similarly